### PR TITLE
option to delete source maps after uploading

### PR DESCRIPTION
**Pull request by carchrae**
addresses #40

### CODE REVIEW

**Comment by jan-swiecki** [comment]
:+1: for this. Much needed feature.

Waiting for merge :crossed_fingers:

**Comment by nspilman** [comment]
Thank you for working on this, @carchrae ! Looking forward to this release

**Comment by carchrae** [comment]
i can see a typo in the doc, but aside from that this has been working well for me.  

however, one minor thing, perhaps inconsequential, is that the obfuscated source code has a reference to the `.map` files, so a developer may see a warning that the `.map` file does not exist.  i suppose in the ideal world, you could rebuild the project without source maps or simply remove this reference - but the extra build time/etc did not make that seem worthwhile to me.

File: index.js
Comment (leeandher) [inline index.js:95]: Since the `resolve` values aren't being used for anything, it might be a bit simpler with something like this:
```js
if (inputs.deleteSourceMaps) {
  console.log("Removing source map files.");
  const files = glob.sync(`${sourceMapPath}/**/*.map`);
  const deleteFileErrorHandler = (e) => console.log("Error deleting source map file:", e);
  const deleteFilePromises = files.map((file) =>
    deleteFile(file).catch(deleteFileErrorHandler)
  );
  await Promise.all(deleteFilePromises);
}
```

Comment (leeandher) [reply index.js:95]: This also would remove the "Source map files retained." log, which I feel is a bit redundant.

**Review by leeandher — CHANGES_REQUESTED** [review]
Testing this locally, It worked as expected, but you're right that it results in console warnings.

```
Source map error: Error: request failed with status 404
Resource URL: https://xxx.netlify.app/_next/static/css/abc123.css
Source Map URL: abc123.css.map
```

If it's a limitation of the plugin, I think it's worth adding as a foot note to the README

**Comment by nathan-knight** [comment]
@carchrae are you still interested in working on this? If not I can fork your branch and make the changes that were requested.

**Comment by carchrae** [comment]
thanks for the reminder @nathan-knight - i'll address these right now

File: package.json
Comment (carchrae) [reply package.json]: good point; updated to the more recent/maintained glob

File: index.js
Comment (carchrae) [reply index.js]: agree, your version is simpler.  old glob package didn't have the sync option, hence the promise/resolve/etc.

File: README.md
Comment (carchrae) [reply README.md]: fair point, keep the sourcemap options together.

**Comment by carchrae** [comment]
tested the changes on our deployment, seems ok

**Review by leeandher — APPROVED** [review]
Looks good to me! I'm no longer directly working on the integrations team though, so I will ping @getsentry/ecosystem to merge and release.

**Comment by hloehrmann** [comment]
Maybe recommend using `hidden-source-maps` when using `deleteSourceMaps` so there won't be a broken reference in the obfuscated source.

**Comment by nathan-knight** [comment]
@leeandher is there any chance you could please ping the integrations team again?

**Comment by brianthi** [comment]
Hey folks, appreciate the patience with this ticket. I'll prioritize this for the team to run one final round of testing before merge and release.  We'll target late December and will update here when ready.

**Comment by nathan-knight** [comment]
> Hey folks, appreciate the patience with this ticket. I'll prioritize this for the team to run one final round of testing before merge and release. We'll target late December and will update here when ready.

Hey I hate to be a bother but my team really needs this feature before we can go live with Sentry in production and this PR has been open since 2021.

**Comment by carchrae** [comment]
i've been using this since i opened the PR, just follow these steps: https://github.com/getsentry/sentry-netlify-build-plugin/issues/40#issuecomment-984523460

the only outstanding issue is https://github.com/getsentry/sentry-netlify-build-plugin/pull/50#pullrequestreview-1008249842 but that is really only an issue for developers looking at the console.  no impact on users, so not a priority for me.  you could try the suggestions above to fix that - if you do please report back on clear instructions how to avoid the warning.  

all of the change/review requests on this PR have been pretty trivial.  i say go ahead and add your own plugin directory to your project and deploy it.

**Comment by brianthi** [comment]
We're wrapping up a project that came up in December, and someone will be able to pick this up at the beginning of March.

**Comment by lobsterkatie** [comment]
@carchrae - Thanks for the contribution! I've modified it slightly to use `rimraf`, since it will handle the recursion for us, and pushed it as https://github.com/getsentry/sentry-netlify-build-plugin/pull/76.

Cheers!

**Pull request closed without merging**
